implementations/micrometer-registry-timestream2/src/main/java/io/micrometer/timestream2/TimestreamNamingConvention.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2020 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.timestream2;
+
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.util.StringUtils;
+import io.micrometer.core.lang.Nullable;
+
+/**
+ * {@link NamingConvention} for TimeStream.
+ *
+ * @author Guillaume Hiron
+ * @since 1.6.0
+ */
+public class TimestreamNamingConvention implements NamingConvention {
+
+    //https://docs.aws.amazon.com/timestream/latest/developerguide/API_Record.html
+    private static final int MAX_TAG_KEY_LENGTH = 255;
+    private static final int MAX_TAG_VALUE_LENGTH = 2047;
```

Comment:
Aren't these 256 and 2048 respectively?
Truncate should not have an off by one error and should truncate to this amount of characters.

(There could be another issue with truncate though around `String.length()` I'm going to investigate it and open an issue if needed but this is not in scope for this PR.)

Reply:
update to 256 and 2048